spec/controllers/v1/sessions_controller_spec.rb

```diff
@@ -363,6 +363,133 @@ def expect_logger_msg(level, msg)
         end
       end
 
+      context 'loa1_user' do
+        let(:saml_user_attributes) { loa1_user.attributes.merge(loa1_user.identity.attributes) }
+
+        before { allow(SAML::User).to receive(:new).and_return(saml_user) }
+
+        context 'when user has not accepted the current terms of use' do
+          let(:terms_of_use_agreement_loa3_user) { nil }
+
+          context 'and authentication occurred with a application in TERMS_OF_USE_ENABLED_CLIENTS' do
+            let(:application) { SAML::URLService::TERMS_OF_USE_ENABLED_CLIENTS.first }
+            let!(:tracker) { SAMLRequestTracker.create(uuid: login_uuid, payload: { type: 'idme', application: }) }
+            let(:expected_redirect_url) do
+              'http://127.0.0.1:3001/terms-of-use?redirect_url=http%3A%2F%2F127.0.0.1%3A3001%2Fauth%2Flogin%2Fcallback'
+            end
+
+            it 'redirects to terms of use page' do
+              expect(post(:saml_callback)).to redirect_to(expected_redirect_url)
+            end
+          end
+
+          context 'and authentication occurred with an application not in TERMS_OF_USE_ENABLED_CLIENTS' do
+            let(:application) { 'mhv' }
```

Comment:
❓ MHV appears to be in the `SAML::URLService::TERMS_OF_USE_ENABLED_CLIENTS` constant. Is this spec valid?